packages/angular_devkit/build_angular/src/angular-cli-files/utilities/index-file/augment-index-html.ts

```diff
@@ -246,3 +265,38 @@ function _generateSriAttributes(content: string) {
     { name: 'crossorigin', value: 'anonymous' },
   ];
 }
+
+/**
+ * Safari 10.1 and iOS Safari 10.3 supports modules, but does not support the `nomodule` attribute
+ * they will load <script nomodule> anyway. This snippet solve this problem, but only for script
+ * tags that load external code, e.g.: <script nomodule src="nomodule.js"></script>
+ *
+ * Again: this will **not** prevent inline script, e.g.:
+ * <script nomodule>alert('no modules');</script>.
+ *
+ * This workaround is possible because Safari supports the non-standard 'beforeload' event.
+ * This allows us to trap the module and nomodule load.
+ *
+ * Note also that `nomodule` is supported in later versions of Safari - it's just 10.1 that
+ * omits this attribute.
+ */
+const _noModulePolyfill = tags.oneLine`
```

Comment:
Inline scripts are not a recommended approach from a security perspective.  They also increase the complexity of CSP configurations.  Can you switch this to a file based script?

Reply:
As discussed offline, there might be a way, by leveraging the ScriptsPlugins. Will give it a shot tomorrow.